astropy/cosmology/core.py

```diff
@@ -777,6 +791,46 @@ def nu_relative_density(self, z):
 
         return prefac * self._neff_per_nu * rel_mass
 
+    def _get_nufunc_scalar(self):
+        """ Return the version of the nu relative density function
+        to use in integrals."""
+
+        if self._massivenu:
+            return self._massivenu_relative_density_scalar
+        else:
+            return lambda x : 0.22710731766 * self._Neff
+
+    def _massivenu_relative_density_scalar(self, z):
+        """ Neutrino density function relative to the energy density in
+        photons for massive neutrinos.
+
+        Parameters
+        ----------
+        z : scalar
+           Redshift
+
+        Returns
+        -------
+         f : float
+           The neutrino density scaling factor relative to the density
+           in photons at the specified redshift
+
+        Notes
+        -----
+        This version is specialized to scalar z for performance reasons.
+        It also assumes there are actually massive neutrinos present.
+        """
+
+        # See nu_relative_density for an explanation of the formula
+        # Bizarrely(?), it's -much- faster if _nu_y is available as a
+        # python list instead of a numpy ndarray manipulated using
+        # np.sum, etc.
+        rel_mass = float(self._nmasslessnu)
+        k = 0.3173 / (1.0 + z)
+        for curr_nu_y in self._nu_ylist:
+            rel_mass += ((1.0 + (k * curr_nu_y) ** 1.83)**0.5464480874316939)
+        return 0.22710731766 * self._neff_per_nu * rel_mass
```

Comment:
+1 to defining these in comments - if Astropy is still around in 10 years and someone needs to maintain this, they may need to know how to derive these.